testnet/stacks-node/src/run_loop/neon.rs

```diff
@@ -313,7 +317,30 @@ impl RunLoop {
         // prepare to fetch the first reward cycle!
         target_burnchain_block_height = burnchain_height + pox_constants.reward_cycle_length as u64;
 
+        let should_keep_running = Arc::new(AtomicBool::new(true));
+        let keep_running_writer = should_keep_running.clone();
+
+        termination::set_handler(move || {
+            keep_running_writer.store(false, Ordering::SeqCst);
+        })
+        .expect("Error setting termination handler");
+
         loop {
+            // Orchestrating graceful termination
+            if !should_keep_running.load(Ordering::SeqCst) {
+                info!("Terminating process");
+                // Terminate chains coordinator
+                info!("Stopping chains-coordinator");
+                coordinator_senders.stop_chains_coordinator();
+                // Terminate relayer
+                info!("Stopping relayer");
+                node.terminate_relayer();
+                // Commands handling is async, we'll sleep to make sure
+                // that the exit commands sent are ingested.
+                sleep(Duration::from_secs(10));
```

Comment:
It'd be better to be able to join the thread handles here, so that we don't have to sleep and hope that the threads closed.

Reply:
Seconded.  Definitely want to join threads.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2433/commits/b2cc9e3b1fb15402af9191467a5197a3a160c69d